tests/node-tests/basic-node-tests.js

```diff
@@ -39,8 +39,23 @@ fluid.require("test-module", require, "test-module");
 
 fluid.setLogging(true);
 
+// Set boolean for the  presence of the --tap command line option to output a
+// TAP report for consumption by testem when used as a custom launcher
+var shouldOutputTAP = process.argv.findIndex(function (argument) {
+    return argument.indexOf("--tap") > -1;
+}) > -1;
+
+// For accumulating TAP messages for parsing by testem custom launcher
+var tapOutput = "";
+
 QUnit.testDone(function (data) {
-    fluid.log("Test concluded - " + data.name + ": " + data.passed + " passed, " + data.failed + " failed");
+    var testMessage = "Test concluded - " + data.name + ": " + data.passed + " passed, " + data.failed + " failed";
+    fluid.log(testMessage);
+    if (shouldOutputTAP) {
```

Comment:
I think this is ok. We have long-standing modularisation issues with respect to node-jqUnit, and clearly a lot of this code belongs there - but this probably isn't the time to deal with them - we should wait until we can get dug in to do a full "monorepo" style reorganisation of the Infusion repository and then fold node-jqUnit into this structure as a further "internal module". 

I think the factoring could be improved slightly to stick the TAP output into a separate callback applied to QUnit.testDone, but in general I think it's ok to have such stuff duplicated between basic-node-tests and node-jqUnit until we can have a more thorough refactoring. Please could you add a comment here referencing https://issues.fluidproject.org/browse/FLUID-5744 as a pointer to this work.